posthog/temporal/batch_exports/monitoring.py

```diff
@@ -0,0 +1,220 @@
+import datetime as dt
+import json
+from dataclasses import dataclass
+from uuid import UUID
+
+from temporalio import activity, workflow
+from temporalio.common import RetryPolicy
+
+from posthog.batch_exports.service import (
+    aget_active_event_batch_exports,
+    aupdate_expected_records_count,
+)
+from posthog.batch_exports.sql import EVENT_COUNT_BY_INTERVAL
+from posthog.temporal.batch_exports.base import PostHogWorkflow
+from posthog.temporal.common.clickhouse import get_client
+
+
+class NoActiveBatchExportsFoundError(Exception):
+    """Exception raised when no active events batch exports are found for a given team."""
+
+    def __init__(self, team_id: int):
+        super().__init__(f"No active events batch exports found for team {team_id}")
+
+
+class NoValidBatchExportsFoundError(Exception):
+    """Exception raised when no valid events batch exports are found for a given team."""
+
+    def __init__(self, message: str = "No valid events batch exports found for team"):
+        super().__init__(message)
+
+
+@dataclass
+class BatchExportMonitoringInputs:
+    """Inputs for the BatchExportMonitoringWorkflow.
+
+    Attributes:
+        team_id: The team id to monitor batch exports for.
+    """
+
+    # TODO - make this a list of team ids or single?
+    # or maybe a (list of) batch export id(s) instead?
+    team_id: int
+
+
+@dataclass
+class BatchExportDetails:
+    id: UUID
+    interval: str
+    exclude_events: list[str]
+    include_events: list[str]
+
+
+@activity.defn
+async def get_batch_export(team_id: int) -> BatchExportDetails:
+    """Get the number of records completed for a given team."""
+    models = await aget_active_event_batch_exports(team_id)
+    if len(models) == 0:
+        raise NoActiveBatchExportsFoundError(team_id)
+    if len(models) > 1:
+        activity.logger.warning("More than one active events batch export found; using first one...")
+    model = models[0]
+    if model.interval_time_delta != dt.timedelta(minutes=5):
+        raise NoValidBatchExportsFoundError(
+            "Only batch exports with interval of 5 minutes are supported for monitoring at this time."
+        )
```

Comment:
Ah, good, we are only doing 5 min. Then query is good :+1: Sorry, got to this part later.